policybot/pkg/gh/accum_commit.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gh
+
+// Commit all the accumulated objects to the cache and to durable storage.
+// Note that if any errors occur in the middle of this, partial updates to the
+// cache and DB will happen. Sorry 'bout that. Either way, the accumulator is also
+// clean upon return so it can be reused
+func (a *Accumulator) Commit() error {
+	var err error
+	if err = a.commitUsers(); err == nil {
```

Comment:
Nit: rename err to hr and a method called SUCCEEDED that takes hr and returns a boolean. That will bring this closer to the COM-like programming model. :P